### PR TITLE
APR-908 added json and code to deal with contains within a question tag

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs [hunk 1]
```diff
@@ -120,10 +120,28 @@ protected Next GetNextAction(Page page, List<Answer> answers, ApplicationSection
...
                                 someConditionsNotSatisfied = true;
                             }
+                            else
+                            {
+                                if (!string.IsNullOrEmpty(condition.MustEqual) && questionTag.Value<string>() != condition.MustEqual)
+                                {
+                                    someConditionsNotSatisfied = true;
+                                }
+
+                                if (!string.IsNullOrEmpty(condition.Contains))
```
Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs]: this should be an else if or a switch statement

Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs]: Can you add a couple of unit tests into SetPageAnswersTests

SetAnswersBase is used in quite a few places

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs [hunk 2]
```diff
@@ -120,10 +120,28 @@ protected Next GetNextAction(Page page, List<Answer> answers, ApplicationSection
...
 
-                            if (questionTag == null || questionTag.Value<string>() != condition.MustEqual)
+
+                            if (questionTag == null)
                             {
                                 someConditionsNotSatisfied = true;
                             }
+                            else
+                            {
+                                if (!string.IsNullOrEmpty(condition.MustEqual) && questionTag.Value<string>() != condition.MustEqual)
```
Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs]: You might need to look at line 65 and do something similar there too: GetCheckboxListMatchingNextActions

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs [hunk 3]
```diff
@@ -120,10 +120,28 @@ protected Next GetNextAction(Page page, List<Answer> answers, ApplicationSection
...
+                                        .Split(",", StringSplitOptions.RemoveEmptyEntries);
+                                    if (!listOfAnswers.Contains(condition.Contains))
+                                    {
+                                        someConditionsNotSatisfied = true;
+                                    }
+                                }
+                            }
                         }
                         else
                         {
```
Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs]: Do you need to do similar logic for QuestionId conditions?

File: src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs]: I have now made that change

**Pull request closed without merging**